src/tests/mod.rs

```diff
@@ -15,5 +19,30 @@ struct TestCase<'a> {
     /// The stapled OCSP response given to us by Rustls, if any.
     pub stapled_ocsp: Option<&'a [u8]>,
 
-    pub expected_result: Result<(), rustls::Error>,
+    pub expected_result: Result<(), TlsError>,
+
+    /// An error that should be present inside an expected `CertificateError::Other` variant.
+    ///
+    /// Set this if the error being tested uses `CertificateError::Other` and not statically known
+    /// variants in [TlsError]
+    #[allow(dead_code)]
```

Comment:
on a lark I tried to replace this with a `cfg` directive that excluded the field from the MacOS builds that weren't using it. That ends up just shifting the problem as the `E` generic parameter isn't used at that point and we'd need a `PhantomData` field or some other reworking.

TLDR: It felt to me like `allow(dead_code)` was the right tool for this particular instance.

Reply:
Yup, I agree 👍. Especially in test code I prefer to just silence dead code warnings if the alternative ends up being significantly more complex. [Here](https://github.com/rustls/rustls-platform-verifier/blob/main/src/tests/ffi.rs#L132) is an example of that, where figuring out exact combinations of test-only feature flags wasn't worth the noise.